test/access-token-provider.spec.ts

```diff
@@ -0,0 +1,36 @@
+import assert from "assert";
+import { AccessTokenProvider } from "../api/auth/access-token-provider";
+
+describe("AccessTokenProvider", () => {
+    const staticToken = "test.jwt.token";
+
+    it("should return the static token response from authenticate", async () => {
+        const provider = new AccessTokenProvider(staticToken, "Bearer");
+        const response = await provider.authenticate();
+
+        assert.strictEqual(response.accessToken, staticToken);
+        assert.strictEqual(response.tokenType, "Bearer");
+        assert.strictEqual(response.expiresIn, Number.MAX_SAFE_INTEGER);
```

Comment:
We must not even test these methods. It is an internal implementation (for API key and access token).

Reply:
Reworked.